Public/Src/Cache/VerticalStore/Compositing/CompositingCacheFactory.cs

```diff
@@ -107,5 +109,29 @@ private sealed class Config
                 throw;
             }
         }
+
+        /// <inheritdoc />
+        public IEnumerable<Failure> ValidateConfiguration(ICacheConfigData cacheData)
+        {
+            Contract.Requires(cacheData != null);
+
+            var possibleCacheConfig = cacheData.Create<Config>();
+            if (!possibleCacheConfig.Succeeded)
+            {
+                return new[] { possibleCacheConfig.Failure };
+            }
+
+            Config compositingConfig = possibleCacheConfig.Result;
+
+            var metadataCacheFailures =
+                CacheFactory.ValidateConfig(compositingConfig.MetadataCache)
+                    .Select(failure => new Failure<string>($"{nameof(compositingConfig.MetadataCache)} validation failed", failure));
+
+            var casCacheFailures =
+                CacheFactory.ValidateConfig(compositingConfig.CasCache)
+                    .Select(failure => new Failure<string>($"{nameof(compositingConfig.CasCache)} validation failed", failure));
+
+            return metadataCacheFailures.Concat(casCacheFailures).ToArray();

```

Comment:
>heFailures.Concat(casCacheFai [](start = 30, length = 29)

since you are looping twice here, probably a bit more effecient to just createa list and do addrange twice.

Reply:
and since the validateconfig retuns an array you can pre-compute the size of the errors array to allocalte.
If this code is not perf-sensistive i.e. just boot fine to keep more readable.

---
In reply to: [286711799](https://github.com/microsoft/BuildXL/pull/362#discussion_r286711799) [](ancestors = 286711799)